src/utils/regexp.ts

```diff
@@ -1,5 +1,33 @@
+import { Context } from "../api";
 import { assert } from "./errors";
 
+
+/**
+ * Wrapper around RegExp.exec that takes into account the user's smart case
+ * preference.
+ */
+export function smartExec(re: RegExp, string: string): RegExpExecArray | null {
+  const pattern = re.source;
+  const flags = re.flags;
+
+  if (!Context.current.extension.isSmartCaseEnabled) {
+    return re.exec(string);
+  }
```

Comment:
nit: move at the start of the function to bring the assignments above closer to their usage